### ISSUE TITLE
test script has an issue 

### ISSUE THREAD

**Issue by wuqixuan** [coreos/fleet#1299]
test script has an issue, which cause not able to run gofmt for all package.

**Comment by bcwaldon** [comment]
@wuqixuan what specific behavior are you seeing?

**Comment by wuqixuan** [comment]
Before modification, the gofmt and go vet part will not be execuated as per the output. 

```
root@wood-ubuntu-2:/home/lightos/coreos/fleet/src/mygit/fleet# ./test 
Building fleetd... 
Building fleetctl... 
Running tests...  
ok      github.com/coreos/fleet/agent   0.010s  coverage: 70.6% of statements
ok      github.com/coreos/fleet/api     0.012s  coverage: 80.4% of statements
......
ok      github.com/coreos/fleet/systemd 0.004s  coverage: 15.5% of statements
ok      github.com/coreos/fleet/unit    0.006s  coverage: 88.4% of statements
```

But the test shell is below:
```
# Invoke ./cover for HTML output
COVER=${COVER:-"-cover"}

source ./build
...... 
# split TEST into an array and prepend REPO_PATH to each local package
split=(${TEST// / })
TEST=${split[@]/#/${REPO_PATH}/}

echo "Running tests..."
go test ${COVER} $@ ${TEST}

echo "Checking gofmt..."
fmtRes=$(gofmt -l $FMT)
if [ -n "${fmtRes}" ]; then
        echo -e "gofmt checking failed:\n${fmtRes}"
        exit 255
fi

echo "Checking govet..."
vetRes=$(go vet $TEST)
if [ -n "${vetRes}" ]; then
        echo -e "govet checking failed:\n${vetRes}"
        exit 255
fi

echo "Success"
```
Obviously, after "go test $COVER $@ $TEST", the shell exit. So "gofmt -l $FMT" and "go vet $TEST" cannot be executed. 
That's why I move "go test" after "gofmt -l $FMT" and "go vet $TEST".

**Comment by bcwaldon** [comment]
If we invert the order of operations here, however, we have the same problem. `go test` won't be run if `gofmt` or `go vet` fail... Are you asserting that we should not run tests if fmt or vet fail?

**Comment by wuqixuan** [comment]
Suggest that if gofmt and go vet fail, we end up running the script without calling go test.  That can force people to ensure the correct of gofmt and go vet, that benefits  the quality of the code. 

Actually, currently go test end up the script always even if go test succeeds, that is why I move it to the last.

**Comment by bcwaldon** [comment]
Instead of changing the order, why don't we just track the success of each command (go test, go vet, gofmt), and exit with 1 after attempting all if any fail?

**Comment by wuqixuan** [comment]
I got the reason why go test directly exit the whole shell, that's because of "#/bin/bash -e". So the original algorithm is any of three, that is "go test" and "gofmt" and "govet", is failing, the script will exit. 

In some environments, go test ssh package always fails, but it's not important. So it's better modify the test script to your new logic, that just to track the success of each command (go test, go vet, gofmt), and exit with 1 after attempting all if any fail. 

I have updated the script as per the algorithm. The new algorithm is below:
1. If fleetd or fleetctl fail, the whole shell exit.
2. Always do "go test" and "gofmt" and "govet". Exit with 1 after attempting all if any fail.
